spikesorters/ironclust/ironclust.py

```diff
@@ -1,159 +1,163 @@
+            txt += '{}={}\n'.format(key0, val0)
         txt += 'samplerate={}\n'.format(samplerate)
-        txt += 'detect_sign={}\n'.format(p['detect_sign'])
-        txt += 'adjacency_radius={}\n'.format(p['adjacency_radius'])
-        txt += 'detect_threshold={}\n'.format(p['detect_threshold'])
-        txt += 'merge_thresh={}\n'.format(p['merge_thresh'])
-        txt += 'freq_min={}\n'.format(p['freq_min'])
-        txt += 'freq_max={}\n'.format(p['freq_max'])
-        txt += 'pc_per_chan={}\n'.format(p['pc_per_chan'])
-        txt += 'prm_template_name={}\n'.format(p['prm_template_name'])
-        txt += 'fGpu={}\n'.format(p['fGpu'])
-        _write_text_file(dataset_dir / 'argfile.txt', txt)
-
-    def _run(self, recording, output_folder):
-        if self.debug:
-            print('Running IronClust...')
-
-        dataset_dir = (output_folder / 'ironclust_dataset').absolute()
-
-        cmd_path = "addpath('{}', '{}/matlab', '{}/mdaio');".format(IronclustSorter.ironclust_path, IronclustSorter.ironclust_path, IronclustSorter.ironclust_path)
-        # "p_ironclust('$(tempdir)','$timeseries$','$geom$','$prm$','$firings_true$','$firings_out$','$(argfile)');"
-        cmd_call = "p_ironclust('{}', '{}', '{}', '', '', '{}', '{}');" \
-            .format(output_folder, dataset_dir / 'raw.mda', dataset_dir / 'geom.csv', output_folder / 'firings.mda',
-                    dataset_dir / 'argfile.txt')
-        cmd = 'matlab -nosplash -nodisplay -r "{} {} quit;"'.format(cmd_path, cmd_call)
-        if self.debug:
-            print(cmd)
-
-        if "win" in sys.platform:
-            cmd_list = ['matlab', '-nosplash', '-nodisplay', '-wait',
-                        '-r', '{} {} quit;'.format(cmd_path, cmd_call)]
-        else:
-            cmd_list = ['matlab', '-nosplash', '-nodisplay',
-                        '-r', '{} {} quit;'.format(cmd_path, cmd_call)]
-
-        _call_command_split(cmd_list)
+        with open(dataset_dir / 'argfile.txt', 'w') as f:
+            f.write(txt)
+
+        tmpdir = output_folder / 'tmp'
+        os.mkdir(tmpdir)
+        print('Running ironclust in {tmpdir}...'.format(tmpdir=str(tmpdir)))
+        cmd = '''
+            addpath('{source_dir}');
+            addpath('{ironclust_path}', '{ironclust_path}/matlab', '{ironclust_path}/matlab/mdaio');
+            try
+                p_ironclust('{tmpdir}', '{dataset_dir}/raw.mda', '{dataset_dir}/geom.csv', '', '', '{tmpdir}/firings.mda', '{dataset_dir}/argfile.txt');
+            catch
+                fprintf('----------------------------------------');
+                fprintf(lasterr());
+                quit(1);
+            end
+            quit(0);
+        '''
+        cmd = cmd.format(ironclust_path=IronClustSorter.ironclust_path, tmpdir=str(tmpdir), dataset_dir=str(dataset_dir), source_dir=str(source_dir))
+
+        matlab_cmd = ShellScript(cmd, script_path=str(tmpdir / 'run_ironclust.m'), keep_temp_files=True)
+        matlab_cmd.write()
+
```

Comment:
We have a delete_output_folder attribute for all sorter that could be used here for "keep_temp_files" no ?

Reply:
Yep. I will remove that destracting keep_temp_files. It doesn't apply here.